### ISSUE TITLE
`_geoBoundingBox` filter returns no results when `lat` and `lng` are indexed as strings

### ISSUE THREAD

**Issue by martinmaillard** [meilisearch/meilisearch#3973]
**Describe the bug**

When indexing documents with a `_geo` key containing `lat` and `lng` as string values, the `_geoBoundingBox` filter never returns any resuls. Reindexing the same documents with `lat` and `lng` as `float` values solves the issue.

I'm reporting this as a bug because [the documentation states that `string` values are valid for `lat` and `lng`](https://www.meilisearch.com/docs/learn/fine_tuning_results/geosearch#preparing-documents-for-location-based-search).

> When using JSON and NDJSON, _geo must contain an object with two keys: lat and lng. Both fields must contain either a floating point number or a string indicating, respectively, latitude and longitude:

Note that `_geoRadius` works as expected.

**To Reproduce**
1. Index the following documents: `[{"id": 1, "_geo": {"lng": "6", "lat": "46"}}, {"id": 2, "_geo": {"lng": 6, "lat": 46}]`
2. Query the index with the following filter: `{"filter": "_geoBoundingBox([47, 7], [44, 4])"}`
3. Only the document with `id=2` is returned

**Expected behavior**
Both documents should be returned

**Meilisearch version:**
Tested with 1.2 and 1.3

------

## Comment from the team:
This issue was an indexing issue. The fix will be shipped in a patch version which means you don't need a dump to use it.
For all the new documents you send, it'll take effect immediately BUT, if your database already contains geo documents with their coordinates as string, you won't be able to query them until you re-index them by removing and then adding again `_geo` as a faceted attribute (i.e.: removing it from both the sortable and filterable attributes).
You can also delete and send back all the affected documents if that's easier for you.

**Comment by irevoire** [comment]
Hey @martinmaillard.

It seems like you mixed up a bit the order of your latitude/longitude.

[As you can see in the documentation](https://www.meilisearch.com/docs/learn/fine_tuning_results/geosearch#_geoboundingbox), the `_geoBoundingBox` function must receive the latitude and then the longitude, i.e.: `_geoBoundingBox([{lat}, {lng}], [{lat}, {lng}])`.
And as stated later:
> The first array indicates the geographic coordinates of the top right corner of the rectangular area. The second array indicates the coordinates of the bottom left corner of the rectangular area.

Following these indications, this means the top right corner should be expressed as `[47, 7]`, and your bottom left corner as `[44, 6]`.
Finally, for your documents, the final filter you wanted to send was this one: `{ "filter": "_geoBoundingBox([47,7], [44,6])" }`.

**Comment by martinmaillard** [comment]
Hi @irevoire,

Sorry, I don't understand your message. Where did I mix up `lat` and `lng`? The filter you wrote looks exactly like mine, except it has a `lng` of `6` instead of `4`, which shouldn't change anything in this case.

Am I missing something?

**Comment by irevoire** [comment]
Oh yes you’re right, I’m the one who was confused 😬 

I added the document you shared and ran your exact search, and it worked on my side 🤔 

Maybe you could share more info on what you did? Or send me a dump of your instance if that’s possible?

Here are the steps I ran that worked for me:
```
% echo '[{"id": 1, "_geo": {"lng": "6", "lat": "46"}}, {"id": 2, "_geo": {"lng": 6, "lat": 46}}]' | mieli document add
{
  "taskUid": 104,
  "indexUid": "mieli",
  "status": "enqueued",
  "type": "documentAdditionOrUpdate",
  "enqueuedAt": "2023-08-07T11:01:31.960258Z"
}
  succeeded
{
  "uid": 104,
  "indexUid": "mieli",
  "status": "succeeded",
  "type": "documentAdditionOrUpdate",
  "canceledBy": null,
  "details": {
    "receivedDocuments": 2,
    "indexedDocuments": 2
  },
  "error": null,
  "duration": "PT0.900846S",
  "enqueuedAt": "2023-08-07T11:01:31.960258Z",
  "startedAt": "2023-08-07T11:01:31.962533Z",
  "finishedAt": "2023-08-07T11:01:32.863379Z"
}
% echo '{ "filterableAttributes": ["_geo"] }' | mieli settings
{
  "taskUid": 105,
  "indexUid": "mieli",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2023-08-07T11:10:59.265416Z"
}
  succeeded
{
  "uid": 105,
  "indexUid": "mieli",
  "status": "succeeded",
  "type": "settingsUpdate",
  "canceledBy": null,
  "details": {
    "filterableAttributes": [
      "_geo"
    ]
  },
  "error": null,
  "duration": "PT31.578141S",
  "enqueuedAt": "2023-08-07T11:10:59.265416Z",
  "startedAt": "2023-08-07T11:10:59.266412Z",
  "finishedAt": "2023-08-07T11:11:30.844553Z"
}
% echo '{ "filter": "_geoBoundingBox([47,7], [44,4])" }' | mieli search
```

**Comment by irevoire** [comment]
Oops, yes, you’re right. Sorry, I totally misunderstood your issue 🤦 

The bug is related to the fact that the `_geoBoundingBox` is kind of a hack around the filtering system of meilisearch.
But since the `range` doesn’t work on strings with the meilisearch filter, the `_geoBoundingBox` can't work with the strings either.
This issue is not trivial to fix.

**Comment by martinmaillard** [comment]
Ok, I see. Thanks for looking into it :)

Maybe the fix is to change the documentation of `_geo` to make it clear that `lat` and `lng` can't be strings at the moment. That would probably save time for some people.

**Comment by irevoire** [comment]
Hey @martinmaillard, just to keep you updated.
I chatted about this issue with the team, and we found a cool solution without any downside. It's implemented and will be released in the next patch version of meilisearch (probably by the end of the week or next week).
Also, I’ll update your initial message so it's easier for the rest of the team to follow what's going on!
Thanks for the report 🎉

**Comment by curquiza** [comment]
Fixed by #3986 without downside
Will be released today in v1.3.1
